### PR TITLE
[hailtop/batch] Add logs messages to tests

### PR DESCRIPTION
**Pull request by jigold**
I didn't test the getting the logs code.

### CODE REVIEW

File: docker/third-party/images.txt [hunk 1]
```diff
@@ -5,6 +5,12 @@ amundsendev/amundsen-search:2.4.0
 debian:9.5
 ghost:3.0-alpine
 google/cloud-sdk:305.0.0-slim
+hailgenetics/python-dill:3.6
+hailgenetics/python-dill:3.6-slim
+hailgenetics/python-dill:3.7
+hailgenetics/python-dill:3.7-slim
+hailgenetics/python-dill:3.8
+hailgenetics/python-dill:3.8-slim
```
Comment (danking) [inline docker/third-party/images.txt:13]: this seems a bit odd. These images are defined in docker/python-dill. Maybe we should just add a `gcr.io/..` alias there?

File: hail/python/test/hailtop/batch/utils.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from hailtop.batch_client.client import Job
+
+
+def job_logs(batch):
+    logs = {}
+    jobs = batch.jobs()
+    for j_status in jobs:
+        j = Job(batch, j_status['job_id'])
+        logs[j.id] = j.log()
+    return logs
```
Comment (danking) [inline hail/python/test/hailtop/batch/utils.py:10]: Pytest will truncate the output unless it is a string. Either convert to a string here with `str(logs)` or convert in each assert: `str(job_logs(res))`.

File: hail/python/test/hailtop/batch/test_batch_pool_executor.py [hunk 3]
```diff
@@ -1,30 +1,33 @@
...
 import time
+import os
 from hailtop.batch import BatchPoolExecutor
 
+PYTHON_DILL_IMAGE = os.environ['PYTHON_DILL_IMAGE']
+
 
 def test_simple_map():
-    with BatchPoolExecutor(project='hail-vdc') as bpe:
+    with BatchPoolExecutor(project='hail-vdc', image= PYTHON_DILL_IMAGE) as bpe:
```
Comment (danking) [inline hail/python/test/hailtop/batch/test_batch_pool_executor.py:10]: no spaces after kwargs

**Review by danking — APPROVED** [review]

**Review by danking — CHANGES_REQUESTED** [review]
xxx

File: docker/third-party/images.txt (hunk 1 above)
Comment (jigold) [reply docker/third-party/images.txt:13]: Can you expand on this idea more?

**Review by danking — APPROVED** [review]

**Pull request merged**